File: app/schemas/admin_adoption.py
from datetime import datetime

from pydantic import BaseModel


class AdminAdoptionItemRespuesta(BaseModel):
    usuario_id: int
    profesional_id: int
    cuenta_id: int
    nombre: str
    apellido: str
    email: str
    first_login_at: datetime | None
    last_login_at: datetime | None
    login_count: int
    last_activity_at: datetime | None
    days_since_last_activity: int | None
    active_days: int
    patients_created: int
    appointments_created: int
    services_created: int
    availabilities_created: int
    clinical_evolutions_created: int
    adoption_status: str

File: frontend/src/types/adminAdoption.ts
export type AdoptionStatus = "sin_uso" | "configurando" | "activo" | "en_riesgo" | "inactivo";

export interface AdminAdoptionItem {
  usuario_id: number;
  profesional_id: number;
  cuenta_id: number;
  nombre: string;
  apellido: string;
  email: string;
  first_login_at: string | null;
  last_login_at: string | null;
  login_count: number;
  last_activity_at: string | null;
  days_since_last_activity: number | null;
  active_days: number;
  patients_created: number;
  appointments_created: number;
  services_created: number;
  availabilities_created: number;
  clinical_evolutions_created: number;
  adoption_status: AdoptionStatus;
}

export interface FiltrosAdopcionAdmin { status?: AdoptionStatus; q?: string; }
